lib/libutils/ext/ubsan.c

```diff
@@ -39,6 +43,18 @@ static void __noreturn ubsan_panic(void)
 	while (1);
 }
 
+static bool was_already_reported(struct source_location *loc)
+{
+	uint32_t column;
+
+	column = loc->column;
```

Comment:
Why consider only the column? The file path and the line number are important as well.

Reply:
> Why consider only the column? The file path and the line number are important as well.

The unique source location have 1:1 mapping with `struct source_location *loc `, the last bit of column field only used to store flag, that this source location was already reported. Previously I used union, but it was [requested](https://github.com/OP-TEE/optee_os/pull/7312#discussion_r1995079647) to remove it.

Reply:
Got it. For simplicity I would personally have added a `bool reported;` to `struct source_location` and dropped the atomic swap (who cares if the same location might be reported twice once in a blue moon?) but the code you propose is clearly correct and better, if a bit more complex. So that's fine with me.

Reply:
> Got it. For simplicity I would personally have added a bool reported; to struct source_location and dropped the atomic swap (who cares if the same location might be reported twice once in a blue moon?) but the code you propose is clearly correct and better, if a bit more complex. So that's fine with me.

That is not possible, this structures are created/passed to callbacks by gcc, we can't change their layout. Linux [do](https://github.com/torvalds/linux/blob/master/lib/ubsan.c#L112) the same

Reply:
Done